main.lua

```diff
@@ -40,46 +115,17 @@ local function exit(status)
     end
   end)
   uv.run()
-  os.exit(status)
+  os.exit(exitCode)
 end
 
-_G.p = require('pretty-print').prettyPrint
-local version = require('./package').version
-coroutine.wrap(function ()
-  local log = require('log').log
-  local command = args[1] or "help"
-  if command:sub(1, 2) == "--" then
-    command = command:sub(3)
-  end
-  command = aliases[command] or command
-  local invalid = false
-  local success, err = xpcall(function ()
-    log("lit version", version)
-    log("luvi version", require('luvi').version)
-    if command == "version" then exit(0) end
-    local path = "./commands/" .. command .. ".lua"
-    if bundle.stat(path:sub(3)) then
-      log("command", table.concat(args, " "), "highlight")
-    else
-      invalid = command
-      log("invalid command", command, "failure")
-      command = "help"
-      path = "./commands/" .. command .. ".lua"
-    end
-    require(path)()
-  end, debug.traceback)
-  if invalid then
-    success = false
-    err = "Invalid Command: " .. invalid
-  end
-  if success then
-    log("done", "success", "success")
-    print()
-    exit(0)
-  else
-    log("fail", err, "failure")
-    print()
-    exit(-1)
-  end
-end)()
-uv.run()
+function commandLine.isValidCommand(command)
+  local commandHandler = "./commands/" .. command .. ".lua"
+  return bundle.stat(commandHandler:sub(3)) -- A command is valid if a script handler for it exists
+end
+
+function commandLine.executeCommandHandler(command)
+  local commandHandler = "./commands/" .. command .. ".lua"
+  require(commandHandler)()
+end
```

Comment:
This makes things more readable, but it's not really a refactor since they both have to execute the same code. Maybe build `commandHandler` and then use it in each function?

Reply:
Hmm. Not sure what happened here, but I think the path was generated twice in the original also. TBH I haven't found a great way of removing the redundancy while also keeping it as readable as possible, but I've pushed a change that would at least eliminate the duplicated path computation.